lib/matobj.gi

```diff
@@ -1345,6 +1353,8 @@ InstallMethod( Characteristic,
     [ IsMatrixOrMatrixObj ],
     M -> Characteristic( BaseDomain( M ) ) );
 
+InstallOtherMethod(DefaultFieldOfMatrix,[IsMatrixOrMatrixObj],BaseDomain);
```

Comment:
I have been using `DefaultFiledOfMatrix` in my code to get the definition field of an (ordinary) matrix. How would we get it in a uniform way, regardless whether an object is `IsMatrix' or `IsMatrixObj`. Will `BaseDomain` work for the "classical" matrices?

Reply:
Yes `BaseDomain` works for "classical" matrices (but there was a bug: it only looked .

But since it has different semantics, it is not so clear what the "correct" answer should be for e.g. an identity matrix with integer entries: should it be `Integers` or `Rationals`? This is of course clear with `DefaultFiledOfMatrix`. But `BaseDomain` currently returns `Integers`. We could of course change that, but in the end, it'll always be a guessing game; which is after all one of the primary reasons why we want MatrixObj: so that we can stop guessing and start specifying what we want.